README.rst

```diff
@@ -0,0 +1,227 @@
+FunFlow
+=======
+
+.. highlight:: haskell
+.. default-role:: code
+
+.. image:: https://circleci.com/gh/tweag/funflow.svg?style=svg
+    :target: https://circleci.com/gh/tweag/funflow
+
+FunFlow is a library and tools to compose and run computational workflows. 
+A workflow is a computation built up of multiple steps. These steps are then
+wired together into a larger computation. Outputs from previous steps can form
+inputs to subsequent steps. 
+
+Features
+--------
+
+- Funflow is designed to be **integrated** into your application. It allows for
+  seamless composition of multiple types of computation:
+
+  - Pure functions (of type `a -> b`)
+  - IO actions (of type `a -> IO b`)
+  - *External* computations. External computations are executed outside of the
+    Haskell process. They allow funflow to invoke computations from any
+    language.
+  - User-defined actions. FunFlow is extensible with user-defined actions. By
+    specifying different interpreters for such actions, you can easily test
+    funflow in a mock environment.
+
+- Funflow provides very powerful **caching** for steps.
+
+  - FunFlow's caching is based around the *content store*. Inspired by the nix_
```

Comment:
The `nix` sentence feels like an aside. I'd put it after the benefit, ie lead with "This means Funflow knows when to re-run a computation and when to re-use a previous result."